src/error.rs

```diff
@@ -0,0 +1,128 @@
+use bdk::blockchain::esplora;
+use lightning::ln::msgs;
+use lightning::util::errors;
+use lightning_invoice::payment;
+use std::fmt;
+use std::io;
+use std::time;
+
+#[derive(Debug)]
+/// An error that possibly needs to be handled by the user.
+pub enum LdkLiteError {
+	/// Returned when trying to start LdkLite while it is already running.
+	AlreadyRunning,
+	/// Returned when trying to stop LdkLite while it is not running.
+	NotRunning,
+	/// An input of the funding transaction tried spending a non-SegWit output. This should never happen, but
+	/// better safe than sorry..
+	FundingTxNonWitnessOuputSpend,
+	/// The funding transaction could not be finalized.
+	FundingTxNotFinalized,
```

Comment:
IIUC, it would fail to be finalized if it includes inputs that cannot be satisfied at the current time, e.g., due to being timelocked. Replaced this with a more generic `FundingTxCreationFailed` error.

Reply:
Well, IIUC they go through all previously unsigned inputs, try to derive the right wallet descriptor, and sign them. So in our case you're probably right, but while the word 'finalize' appears quite a lot it's kinda poorly documented what it actually does/why it's an extra step.  I'll try to clarify with the BDK folks and will probably open an issue..

Reply:
So, it seems that while individual inputs are signed before, the witness is just created and added in the finalization step. I'm still not quite sure when this step would ever fail in our case, since not having enough funds should already become apparent during input selection, i.e., during `tx_builder.finish()`. So, I'll probably just capture all that via the `FundingTxCreationFailed` error.